crates/mipsy_lib/src/error/runtime/mod.rs

```diff
@@ -684,6 +698,211 @@ impl Error {
                 error
 
             }
+            Error::InvalidSyscall { syscall , reason } => {
+                // This error is triggered when $v0 is not a known value when `syscall` is used
+                // This error triggers after $v0 is checked for being uninitialize, so we can can assume that $v0 is an integer value
+
+                let syscall = *syscall;
+
+                let mut error = String::new();
+
+                error.push_str("Invalid Syscall\n");
+
+                match reason {
+                    InvalidReason::Unimplemented => {
+                        // $v0 is a valid value, but mipsy doesn't implement the operation
+                        error.push_str(&format!("\nthe syscall number `{}` is not implemented.\n", syscall.to_string().bold()));
+                    }
+                    InvalidReason::Unknown => {
+                        // $v0 is an invalid value, mips doesn't define this syscall
+                        error.push_str(&format!("\nthe syscall number `{}` is no valid.\n", syscall.to_string().bold()));
+                    }
+                }
+
+                let last_mod = get_last_mod(runtime, Register::V0.to_u32());
+                let state = runtime.timeline().state();
+                let inst = state.read_mem_word(state.pc()).unwrap();
+                let decompiled = decompile::decompile_inst_into_parts(binary, inst_set, inst, state.pc());
+
+                // This instruction should always be a `syscall`
+                // This is mainly used to print the line number of the instruction
+                if let ErrorContext::Binary | ErrorContext::Interactive = context {
+                    error.push_str("\nthe instruction that failed was:\n");
+                    error.push_str(&inst_parts_to_string(
+                        &decompiled,
+                        &source_code,
+                        binary,
+                        false,
+                        false,
+                    ));
+                    error.push('\n');
+                }
+
+                let decompiled_all = decompile::decompile_into_parts(binary, inst_set);
+                let decompiled_next = decompile::decompile_inst_into_parts(binary, inst_set, inst, state.pc() + 4);
+                if decompiled.location.is_none() || ( decompiled_all.contains_key(&decompiled_next.addr) && decompiled_next.location.is_none()) {
+                    // The current instruction should always be `syscall`, so we can't get a pseudo-instruction
+                    // Therefor, we don't need to expand it.
+                    unreachable!()
```

Comment:
this argument is faulty -- `syscall` could be *part of* a pseudo-instruction (like in the dbg_print instns, although they cannot fail in this way) -- so this is making an assumption on the set of pseudoinstns

Reply:
That is a very good point.
I will add the normal expansion code.